### PR TITLE
[dev] Update a collection of package management packages

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

An upgrade to libdnf was required to enable other packages. This upgrade ended up pulling in a number of related package upgrades- see below section for details.

###### Change Log  

- Upgrade createrepo_c to 0.17.5
- Upgrade dnf to 4.8.0
- Upgrade file to 5.40
- Upgrade libdnf to 0.63.1
- Upgrade libmodulemd to 2.13.0
- Upgrade librepo to 1.14.2
- Upgrade libsolv to 0.7.19

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local toolchain/package build of affected packages

### CODE REVIEW

File: SPECS/createrepo_c/createrepo_c.spec [hunk 1]
```diff
@@ -39,59 +38,77 @@ Requires:   %{name} = %{version}-%{release}
...
+popd
 
 %files
 %defattr(-, root, root)
 %license COPYING
-/etc/bash_completion.d/createrepo_c.bash
+%{_sysconfdir}/bash_completion.d/createrepo_c.bash
 %{_bindir}/*
-%{_lib64dir}/*.so.*
+%{_libdir}/*.so.*
```
Comment (MateuszMalisz) [inline SPECS/createrepo_c/createrepo_c.spec:75]: Could we add so version numbers to the files list?

Comment (MateuszMalisz) [reply SPECS/createrepo_c/createrepo_c.spec:75]: Same with other so files in other specs

Comment (thcrain-msft) [reply SPECS/createrepo_c/createrepo_c.spec]: Fixed for all specs, thanks

File: SPECS/libdnf/libdnf.spec [hunk 2]
```diff
@@ -87,74 +64,47 @@ Python 3 bindings for the hawkey library.
...
   %make_install
 popd
 
 %find_lang %{name}
 
-%post -p /sbin/ldconfig
-%postun -p /sbin/ldconfig
+%check
+pushd build-py3
+  make ARGS="-V" test
```
Comment (PawelWMS) [inline SPECS/libdnf/libdnf.spec:99]: Neat!

File: SPECS/libmodulemd/libmodulemd.spec [hunk 3]
```diff
@@ -32,47 +33,55 @@ C Library for manipulating module metadata files
...
-       -Dwith_py3_overrides=true -Dwith_py2_overrides=false api1
-cd api1
-ninja
+%meson \
+    -Ddeveloper_build=false \
+    -Dbuild_api_v1=true \
+    -Dbuild_api_v2=true \
+    -Dwith_py3_overrides=true \
+    -Dwith_py2_overrides=false \
+    -Dwith_docs=false \
```
Comment (PawelWMS) [inline SPECS/libmodulemd/libmodulemd.spec:50]: Shouldn't this also now depend on the value of `%{with docs}`?

**Review by PawelWMS — APPROVED** [review]

File: SPECS/libmodulemd/libmodulemd.spec (hunk 3 above)
Comment (thcrain-msft) [reply SPECS/libmodulemd/libmodulemd.spec:53]: Yep, good catch! Fixed.

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**